version/prop.go

```diff
@@ -104,3 +106,76 @@ func initUnstable() {
 	}
 	isUnstableBuild = minor%2 == 1
 }
+
+// Meta is a JSON-serializable type that contains all the version
+// information.
+type Meta struct {
+	// MajorMinorPatch is the "major.minor.patch" version string, without
+	// any hyphenated suffix.
+	MajorMinorPatch string `json:"majorMinorPatch"`
+
+	// IsDev is whether Short contains a -dev suffix. This is whether the build
+	// is a development build (as opposed to an official stable or unstable
+	// build stamped in the usual ways). If you just run "go install" or "go
+	// build" on a dev branch, this will be true.
+	IsDev bool `json:"isDev,omitempty"`
+
+	// Short is MajorMinorPatch but optionally adding "-dev" or "-devYYYYMMDD"
+	// for dev builds, depending on how it was build.
+	Short string `json:"short"`
+
+	// Long is the full version string, including git commit hash(es) as the
+	// suffix.
+	Long string `json:"long"`
+
+	// UnstableBranch is whether the build is from an unstable (development)
+	// branch. That is, it reports whether the minor version is odd.
+	UnstableBranch bool `json:"unstableBranch,omitempty"`
+
+	// GitCommit, if non-empty, is the git commit of the
+	// github.com/tailscale/tailscale repository at which Tailscale was
+	// built. Its format is the one returned by `git describe --always
+	// --exclude "*" --dirty --abbrev=200`.
+	GitCommit string `json:"gitCommit,omitempty"`
+
+	// GitDirty is whether Go stamped the binary has having dirty version
```

Comment:
I think "has having" should be "as having"

Reply:
Hah, I copy/pasted this from elsewhere that was wrong. Fixed in both places.